lib/gamora/authentication/base.rb

```diff
@@ -47,12 +47,26 @@ def user_attributes_from_claims(claims)
 
       def resource_owner_claims(access_token)
         return {} if access_token.blank?
-        oauth_client.userinfo(access_token)
+        resource_owner_claims!(access_token)
+      end
+
+      def resource_owner_claims!(access_token)
+        Rails.cache.fetch(cache_key(access_token), cache_options) do
+          oauth_client.userinfo(access_token)
+        end
       end
 
       def oauth_client
         Client.from_config
       end
+
+      def cache_options
+        { expires_in: Configuration.userinfo_cache_expires_in }
+      end
+
+      def cache_key(access_token)
```

Comment:
there is some weird edge cases where the session id can be `nil`

https://stackoverflow.com/questions/7759175/rails-session-id-is-nil

Can you ensure that we dont end up with a cache key called: `userinfo:` ?

Reply:
Fixed. Digest from access token is being used now.